includes/utils/class-amp-wp-utils.php

```diff
@@ -58,4 +58,28 @@ protected static function _wp_translate_php_url_constant_to_key( $constant ) {
 
 		return false;
 	}
+
+	/**
+	 * Adds 'data-amp-layout' to the allowed <img> attributes for wp_kses() in a 'post' context.
+	 *
+	 * @since 0.7
+	 *
+	 * @param array  $context      Allowed tags and their allowed attributes.
+	 * @param string $context_type Type of context.
+	 * @return array $context      Filtered allowed tags and attributes.
+	 */
+	public static function add_layout( $context, $context_type ) {
+		if ( 'post' !== $context_type ) {
+			return $context;
+		}
```

Comment:
Is it alright to only filter the allowed attributes [in a 'post' context](https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/kses.php#L850)?

Reply:
I'd say it should be allowed in any context where an `img` is allowed with a `width` and `height`.

Reply:
Thanks, [this commit](https://github.com/Automattic/amp-wp/pull/937/commits/a233b59e1d4088eae1f614fc4ed856c700959558) allows `data-amp-layout` whenever an `<img>` permits `width` and `height`.